Automattic/woocommerce-payments
# refactor: settings selector hooks return type

Fixes #

#### Changes proposed in this Pull Request









Just an optimization to remove the dependency on the selectors and ensuring that just the selector's data is returned from `useSelect`. Now each `useSelect` return the data of only one selector.

This ensures that areas of the settings page aren't being refreshed without us wanting to.

Before - you can see that multiple areas of the app are re-rendered on settings update:
![2021-07-14 15 30 21](https://user-images.githubusercontent.com/273592/125689129-df0d6b99-28b1-4d85-8030-48fe599996f4.gif)

After - no longer the case:
![2021-07-14 15 30 37](https://user-images.githubusercontent.com/273592/125689138-797fc2ba-d690-4ba8-b91d-21cbd32fc573.gif)


#### Testing instructions





- Go to http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments
- Settings are still loaded and persisted on save/refresh
- To see the component refreshes: 
  - Install React Dev Tools
  - Open "Inspect element" on the browser
  - Go to the "Components" tab
  - Enable the "Highlight updates when components render" option
![Markup 2021-07-14 at 15 32 54](https://user-images.githubusercontent.com/273592/125689092-1f9b1645-abc6-46ba-aae3-f5885148b117.png)


-------------------
